### DISCUSSION TITLE
How do soybean plants die? Is it possible?

### DISCUSSION THREAD

**Discussion by CatherineGilbert** [APSIMInitiative/ApsimX#9452] [Q&A] [answered]
I'm asking the question because I can't seem to get soybean to die of frost. The plants pause their development over the winter when thermal time ([Soybean].Phenology.ThermalTime) and the fractional growth rate ([Leaf].FRGR) drop to zero. But as soon as the weather warms again, they can continue accumulating thermal time and develop normally. 

Here's an example from a simulation with MG IV (Generic_MG4) soybean planted in Omaha Nebraska on the 1st of September:

![image](https://github.com/user-attachments/assets/3a2c5598-9079-4784-a1b2-e52834e7221d)

Despite the fact that winter temperatures are -20 C, the plants simply stall and continue development in the spring.

It might be that some portion of the total plants are killed and not the entirety, but I can't figure out how to track this. The total mass of dead matter ([Soybean].TotalDead.Wt) only seems to increase under pod maturity / harvest conditions. There's values named [Soybean].MortalityRate and [Soybean].SeedMortalityRate, but they're constants set to zero.

On the side of the leaf model, [Leaf].LAIDead.LeafDeadWt, [Leaf].KDead, [Leaf].Dead.Wt, and [Leaf].CoverDead also don't seem to be affected by temperature. I can't find anyway for frost to decrease LAI, or for senescence to be triggered outside of phenological conditions, so I can't use total senescence before maturity as a tell that the plants are dead. 

So far, I've been unable to totally kill the crop under any circumstances. The only thing that stops the crop is if it reaches the next year's planting date and is replaced with a new crop. 

In the documentation for the older version of APSIM, there was a list of [causes of plant death](
https://www.apsim.info/documentation/model-documentation/crop-module-documentation/plant/).

![image](https://github.com/user-attachments/assets/4219bb30-8e03-4cb4-8315-268adedea7a7)

Are any of these still implemented? Is killing the crop (not temporarily reducing the rate of its growth to zero) just outside the scope of the model?

**Comment by BrianCollinss** [comment, marked as answer]
You can right-click on your plant in your simulation and select "Show Model Structure". Then you can investigate whether these 'death' criteria (from APSIM Classic) are implemented at all. If not, you can create your own Manager Script (or modify the existing Harvest scripts) and implement your own rules (using `Crop.EndCrop()`).

![image](https://github.com/user-attachments/assets/5506225b-71a7-4828-9ab7-883b91dcb62d)

**Reply by CatherineGilbert** [reply]
I wasn't able to find these 'death' conditions in the Soybean model structure, so I've written a Manager Script with some custom rules to end the crop.

**Reply by LenLon** [reply]
Would you be willing to share this script, or at least the general structure? I assume EndCrop() works for all crops, not just soybean :-)

**Reply by CatherineGilbert** [reply]
[cropkill.txt](https://github.com/user-attachments/files/18103052/cropkill.txt)

Sure! Here's the script I'm using.
